app/controllers/courses_controller.rb

```diff
@@ -1,5 +1,5 @@
 class CoursesController < ApplicationController
   def index
-    # @courses = current_user.courses
+    @courses = current_user.courses.includes(:roles)
```

Comment:
Yeah, recognized there was an issue with this but was not 100% certain how to make it better. This makes loads of sense.